### PR TITLE
Conso preserve type

### PR DESCRIPTION
**Pull request by brandonwillard**
Preserve custom tuple and list types in `conso`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,12 +2,12 @@ sudo: false
...
 python:
     - "2.7"
-    - "3.3"
     - "3.4"
     - "3.5"
+    - "3.6"
     - "pypy"
 install:
-    - pip install coverage pytest pytest-cov flake8
+    - pip install coverage pytest "pytest-cov<2.6.0" flake8
```
Comment (lucaswiman) [inline .travis.yml]: Why was this required? Could you add a comment explaining it?

Comment (brandonwillard) [reply .travis.yml]: It's the error from this run: https://travis-ci.org/logpy/logpy/jobs/499555791#L529

Comment (brandonwillard) [reply .travis.yml]: Apparently, `pytest-cov` versions greater than the pinned one will require a higher `pytest` version.  As an alternative, we could try the newer `pytest`.

Comment (lucaswiman) [reply .travis.yml]: Oh I see, so the version of pytest installed on the travis environment is too old for newer versions of pytest-cov. That makes sense. It's fine to leave it as-is.

**Comment by lucaswiman** [comment]
This looks reasonable to me. I'll merge tomorrow and try deploying to pypi.

What are you using kanren for?

**Comment by brandonwillard** [comment]
Sounds good; thanks!

Right now, I'm using it for this project: [`symbolic-pymc`](https://github.com/pymc-devs/symbolic-pymc).  It could definitely use some input from you `logpy` folks&mdash;if only to help fend off a complete transition to [Hy](https://github.com/hylang/hy) and its miniKanren, [`adderall`](https://github.com/algernon/adderall) (and/or my staging version [`hypoKanren`](https://github.com/brandonwillard/hypoKanren)).  

Although, I'm heavily involved with the Hy world and really can't recommend it enough, I'm trying to keep `symbolic-pymc` from having it as a core requirement (for now), so that it's just a little more welcoming to the Python community.  The ambiguities of tuple-based forms/expressions and the complete lack of macros are really hard to deal with, though.
